apiserver/cmd/apiserver/server/watch.go

```diff
@@ -81,6 +84,7 @@ func WatchExtensionAuth(ctx context.Context) (bool, error) {
 					n := new.(*corev1.ConfigMap)
 					// Only detect as changed if the version has changed
```

Comment:
Do we definitely care about every revision change here?  Seems like we could easily check the data for changes instead:
```
if len(n.Data) != len(o.Data) { changed = true ... }
for k, v := range n.Data {
  if ov, ok := o.Data[k] !ok || ov != v { changed = true ... }
}
// Same for n.BinaryData ...
```
(It's common for an operator to tweak annotations on the object every few seconds, but we should look at the data we care about and ignore anything else.)

Reply:
The intention of this change is to see if our cache is stale and re-iterating over the same revisions post-restart, rather than investigating the change itself. Though, I suppose if we confirm that our cache is not stale, inspecting the changes themselves is the next step, so I'll put both in, thanks!

Reply:
This is done now